pkg/reconciler/events/cloudevent/cloudeventsfakeclient.go

```diff
@@ -72,7 +83,67 @@ func (c FakeClient) StartReceiver(ctx context.Context, fn interface{}) error {
 	return nil
 }
 
-// WithClient adds to the context a fake client with the desired behaviour
-func WithClient(ctx context.Context, behaviour *FakeClientBehaviour) context.Context {
-	return context.WithValue(ctx, ceKey{}, newFakeClient(behaviour))
+// addCount can be used to add the count when each event is going to be sent
+func (c FakeClient) addCount() {
+	c.waitGroup.Add(1)
+}
+
+// decreaseCount can be used to the decrease the count when each event is sent
+func (c FakeClient) decreaseCount() {
+	c.waitGroup.Done()
+}
+
+// WithClient adds to the context a fake client with the desired behaviour and expectedEventCount
+func WithClient(ctx context.Context, behaviour *FakeClientBehaviour, expectedEventCount int) context.Context {
+	return context.WithValue(ctx, ceKey{}, newFakeClient(behaviour, expectedEventCount))
+}
+
+// CheckCloudEventsUnordered checks that all events in wantEvents, and no others,
+// were received via the given chan, in any order.
+// Block until all events have been sent.
+// If more events were sent than expected, this will block indefinitely because the buffered channel
+// is initialized to the expected number of events and will not be able to send any more.
+func (c *FakeClient) CheckCloudEventsUnordered(t *testing.T, testName string, wantEvents []string) {
+	t.Helper()
+	c.waitGroup.Wait()
+	err := eventsFromChannelUnordered(c.Events, wantEvents)
+	if err != nil {
+		t.Fatalf("error getting events from fake client: %v", err)
+	}
+}
+
+// eventsFromChannelUnordered takes a chan of string and a list of events that a test
+// expects to receive. The events can be received in any order. Any extra or too few
+// events are both considered errors.
+func eventsFromChannelUnordered(c chan string, wantEvents []string) error {
```

Comment:
It might make sense to merge this function with `CheckCloudEventsUnordered`, since it's not used anywhere else as far as I can tell, and it needs the call to waitgroup.Wait in order to work correctly.

Reply:
👍